src/pool/pool_disjoint.c

```diff
@@ -135,25 +133,32 @@ static void destroy_slab(slab_t *slab) {
         LOG_ERR("deallocation of slab data failed!");
     }
 
-    umf_ba_global_free(slab->chunks);
     umf_ba_global_free(slab);
 }
 
-// return the index of the first available chunk, SIZE_MAX otherwise
 static size_t slab_find_first_available_chunk_idx(const slab_t *slab) {
-    // use the first free chunk index as a hint for the search
-    for (bool *chunk = slab->chunks + slab->first_free_chunk_idx;
-         chunk != slab->chunks + slab->num_chunks_total; chunk++) {
-
-        // false means not used
-        if (*chunk == false) {
-            size_t idx = chunk - slab->chunks;
-            LOG_DEBUG("idx: %zu", idx);
-            return idx;
+    // Calculate the number of 64-bit words needed.
+    size_t num_words = (slab->num_chunks_total + 63) / 64;
+    for (size_t i = 0; i < num_words; i++) {
+        // Invert the word: free bits (0 in the allocated mask) become 1.
+        uint64_t word = ~(slab->chunks[i]);
+
+        // For the final word, clear out bits that exceed num_chunks_total.
+        if (i == num_words - 1) {
```

Comment:
nit: if you would represent free chunk as 1 (instead of 0) you could remove this if I think

Reply:
Or you can set the appropriate bits as allocated during initalization.

Reply:
also, is this really necessary? There's a condition that seemingly checks for the same thing here:
https://github.com/oneapi-src/unified-memory-framework/pull/1147/files#diff-7d792b84303f5dbbfba19a0b300b3645a0455df415f4246568dc0b1a5b846c73R156-R158

If we what Igor or I suggested, we should be able to remove that condition.

Reply:
done